### PR TITLE
Fix #81 - Prevent normalize/referrer from causing referrer problem.

### LINKED ISSUE #81: Uncaught TypeError: Failed to construct 'Request': The origin of 'https://t.co/VTFdY6dHFm' should be same as 'https://davidwalsh.name'
When someone clicks a link from Twitter to my blog, which gets masked by their URL shortener (https://t.co/VTFdY6dHFm), the following error is displayed in the console:
requestcontructerror

### PR DESCRIPTION
**Pull request by darkwing**
This prevents the referrer problem as seen in issue #81 .

### CODE REVIEW

**Comment by darkwing** [comment]
We still see this in the Chrome console:

chromeissue

But IMO it's better than Firefox throwing a fetch error.

**Comment by darkwing** [comment]
@marco-c @delapuente

File: wp-offline-shell/lib/service-worker.js [hunk 1]
```diff
@@ -15,26 +15,6 @@
...
-      url.search = '';
-      url.fragment = '';
-      return new Request(url, {
-        method: request.method,
-        headers: request.headers,
-        mode: 'no-cors',
-        credentials: request.credentials,
-        cache: request.cache,
-        redirect: request.redirect,
-        referrer: request.referrer,
```
Comment (darkwing) [inline wp-offline-shell/lib/service-worker.js]: This appeared to be the line that was causing the Firefox error.

File: wp-offline-shell/lib/service-worker.js [hunk 2]
```diff
@@ -107,14 +87,14 @@
...
-      if (!this.shouldBeHandled(lookupRequest)) {
+      if (!this.shouldBeHandled(request)) {
         return;
       }
 
       event.respondWith(
-        caches.match(lookupRequest)
-          .then(response => {
+        caches.match(request, {
+          ignoreSearch: true
```
Comment (marco-c) [inline wp-offline-shell/lib/service-worker.js]: I think the `ignoreSearch` isn't available in Chrome.

File: wp-offline-shell/lib/service-worker.js [hunk 3]
```diff
@@ -107,14 +87,14 @@
...
-      var lookupRequest = this.normalizeAndAnonymize(request);
-      if (!this.shouldBeHandled(lookupRequest)) {
+      if (!this.shouldBeHandled(request)) {
         return;
       }
 
       event.respondWith(
-        caches.match(lookupRequest)
-          .then(response => {
+        caches.match(request, {
```
Comment (marco-c) [inline wp-offline-shell/lib/service-worker.js]: What if you passed the cleansed URL instead of the request? So we don't have to use `ignoreSearch`.

**Comment by delapuente** [comment]
With this PR, caching x-origin resources is lost. Or it is an intentional change (so in that case we should change the name of the method to something like `normalize()` as we are anonymizing no more) or must we revert this change and simply delete referrer which is a more suitable specific solution.

**Comment by marco-c** [comment]
@delapuente I think this PR hasn't modified the previous behavior for cross-origin requests. We're using the URL of the request to find an element in the cache, and if an element isn't found we're using `fetch` with the original request.

**Comment by delapuente** [comment]
From [this line](https://github.com/mozilla/wp-offline-shell/pull/83/files#diff-02227a546d60834910d1e8e92ae38176R114), I realized we don't reuse the url for fetching, only for cache matching. Then, there is no alteration of the semantics but I would recommend [changing the name of the method](https://github.com/mozilla/wp-offline-shell/issues/86) to not lead to misconceptions #86.

**Pull request merged**